### ISSUE TITLE
'%' character in cookie value is encoded twice

### ISSUE THREAD

**Issue by grainrigi** [3846masa/http-cookie-agent#26]
## Describe the bug

'%' character is always encoded into '%25' due to `cookie.serialize` behavior.

## To Reproduce

Note: For complete reproduction step, please refer the code below.

Consider a response with a `set-cookie` header like this:

```
Set-Cookie: session=s%3ABFVJo3Ue8cl_m2AAowgqardSnOhJ0eF8.7XmB...
```

In the `set-cookie` handler of this library, the cookie value is stored with `jar.setCookie` method by passing cookie string directly.
Using `tough-cookie`, cookie value is stored without any modifications and urlencoded parts are also left intact.

Next, when sending the cookie value from jar, following occurs:

1. urlencoded cookie value is fetched from `jar.getCookies` method
2. `http-cookie-agent` passes the value (and key) to `cookie.serialize` to combine them
3. `cookie.serialize` applies `encodeURIComponent` to the cookie value

As a result, '%' characters in the cookie value are encoded like this:

```
session=s%253ABFV...
```

This is a broken behavior.

## Expected behavior

Cookie values set by `set-cookie` headers are sent without any modification.

## Code that reproduces the bug

```js
const axios = require('axios').default;
const { CookieJar } = require('tough-cookie');
const { wrapper } = require('axios-cookiejar-support');

const jar = new CookieJar();
const client = wrapper(axios.create({ jar }));

let setCookie;
let cookie;

client
  .get('https://grainrigi.net/cookietest/setcookie.php') // set a cookie 'session' with a urlencoded value
  .then(res => {
    setCookie = res.headers['set-cookie'][0];
    return client.get('https://grainrigi.net/cookietest/showcookie.php'); // echo back the cookie header
  })
  .then(res => {
    cookie = res.data;
    // setCookie and cookie should contain the same cookie value
    console.log(setCookie);
    console.log(cookie);
  });
```

## Environments

- OS: ArchLinux x86_64
- Node.js version: 16.11.1
- axios version: 0.24.0
- tough-cookie version: 4.0.0

**Comment by 3846masa** [comment]
Thanks for reporting.
It was fixed at v1.0.3

**Comment by grainrigi** [comment]
Thanks a lot!
